src/main/webapp/script.js

```diff
@@ -168,6 +170,60 @@ function loadProfile() {
   });
 }
 
+//matches.html functions
+function displayMatches() {
+  let userID = getCurrentUserId();
+  if (userID === null) {
+      return;
+  }
+  fetch('/matches?id=' + userID).then(response => response.json()).then((matches) => {
+    console.log(matches);
+    let matchContainer = document.getElementById('matches-container');
+    for (let i = 0; i < matches.length; i++) {
+      matchContainer.appendChild(createCardElement(matches[i]));
+    }
+  });
+}
+
+function createCardElement(userID) {
+  fetch('/user-data?id=' + userID).then(response => response.json()).then((userinfo) => {
+    if (userinfo === null) {
+        return;
+    }
+    var cardDiv = document.createElement("div");
```

Comment:
`var` is "old school" and discouraged ("banned" is too strong) in new code. You should use `let` or `const`.

See here for some details on `var`: https://javascript.info/var

I would like to not introduce them here in this PR, but let's no go crazy and fix any that are already committed. That would be a nice follow up, though.